fineract-provider/src/main/resources/META-INF/spring/securityContext.xml

```diff
@@ -42,19 +42,26 @@
 				method="POST" requires-channel="https" />
 			<intercept-url pattern="/api/*/self/registration/user" access="permitAll"
 				method="POST" requires-channel="https" />
-			<intercept-url pattern="/api/**" access="isFullyAuthenticated()"
+			<intercept-url pattern="/api/*/twofactor" access="isFullyAuthenticated()"
+						   method="GET" requires-channel="https" />
+			<intercept-url pattern="/api/*/twofactor" access="isFullyAuthenticated()"
+						   method="POST" requires-channel="https" />
+			<intercept-url pattern="/api/*/twofactor/validate" access="isFullyAuthenticated()"
+						   method="POST" requires-channel="https" />
+			<intercept-url pattern="/api/**" access="isFullyAuthenticated() and hasAuthority('TWOFACTOR_AUTHENTICATED')"
 				method="GET" requires-channel="https" />
-			<intercept-url pattern="/api/**" access="isFullyAuthenticated()"
+			<intercept-url pattern="/api/**" access="isFullyAuthenticated() and hasAuthority('TWOFACTOR_AUTHENTICATED')"
```

Comment:
for pattern="/api/*/twofactor , why we have only isFullyAuthenticated() access?
 for other pattern like  pattern="/api/** we have access access="isFullyAuthenticated() and hasAuthority('TWOFACTOR_AUTHENTICATED') ?